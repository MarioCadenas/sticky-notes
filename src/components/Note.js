import React from 'react';
import styled from 'styled-components';

const ContainerDiv = styled.div`
  background-color: yellow;
  display: block;
  float: left;
  margin: 20px;
  min-height: 150px;
  width: 200px;
`;
const ButtonContainer = styled.div`
  background-color: #f3f30c;
  height: 20px;
  width: 100%;
`;
const TextContainer = styled.div`
  word-wrap: break-word;
  width: 100%;
`;
const RemoveButton = styled.button`
  background-color: #f3f30c;
  border: none;
  cursor: pointer;
  float: right;
`;
const Text = styled.p`
  padding: 20px;
`;

const Note = ({ text, removeNote, id }) => {
  const handleRemove = () => removeNote(id);

  return (
    <ContainerDiv>
      <ButtonContainer>
        <RemoveButton onClick={handleRemove}>X</RemoveButton>
      </ButtonContainer>
      <TextContainer>
        <Text>{text}</Text>
      </TextContainer>
    </ContainerDiv>
  );
};

export default Note;
