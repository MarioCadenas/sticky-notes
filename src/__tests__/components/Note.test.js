import React from 'react';
import { shallow } from 'enzyme';
import Note from '../../components/Note';

describe('Note component', () => {
  it('should correctly render', () => {
    const wrapper = shallow(<Note text="test" removeNote={() => {}} id="1" />);
    expect(wrapper).toMatchSnapshot();
  });
});