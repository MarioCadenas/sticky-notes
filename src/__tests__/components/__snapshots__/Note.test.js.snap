// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component should correctly render 1`] = `
<styled.div>
  <styled.div>
    <styled.button
      onClick={[Function]}
    >
      X
    </styled.button>
  </styled.div>
  <styled.div>
    <styled.p>
      test
    </styled.p>
  </styled.div>
</styled.div>
`;
